src/batch/src/executor/join/lookup_join.rs

```diff
@@ -62,9 +66,42 @@ pub trait ProbeSideSourceBuilder: Send {
 }
 
 impl<C: BatchTaskContext> ProbeSideSource<C> {
+    fn get_parallel_unit_ids(&self, scan_range: &ScanRange) -> Result<Vec<ParallelUnitId>> {
+        let dist_keys = self
+            .table_desc
+            .dist_key_indices
+            .iter()
+            .map(|&k| k as usize)
+            .collect_vec();
+        let pk_indices = self
+            .table_desc
+            .order_key
+            .iter()
+            .map(|col| self.table_desc.columns[col.index as usize].column_id as usize)
+            .collect_vec();
+
+        let virtual_node = scan_range.try_compute_vnode(&dist_keys, &pk_indices);
+
+        let mapping = match virtual_node {
+            None => self.vnode_mapping.clone(),
```

Comment:
It means we can't find any vnode mapping here? If so, we should throw an error rather than using all vnodes.

Reply:
It means that we can't find the virtual node corresponding to the scan range.